File: Adapter/PHP/MediaAdapter.php
<?php

require_once "IMediaPlayer.php";

class MediaAdapter {
    private IMediaPlayer $mediaPlayer;

    public function __construct(IMediaPlayer $mediaPlayer)
    {
        $this->mediaPlayer = $mediaPlayer;
    }

    public function playMovie(string $filename) {
        $filename_detail = explode(".", strtolower($filename));
        $count_arr = count($filename_detail);

        if ($count_arr < 2) {
            echo "File not valid\n";
            return;
        }

        switch ($filename_detail[$count_arr - 1]) {
            case 'mkv':
                $this->mediaPlayer->playMKV($filename);
                break;

            case 'mp4':
                $this->mediaPlayer->playMP4($filename);
                break;
            
            default:
                echo "File type not supported\n";
                break;
        }
    }
}

File: Adapter/Python 3/MediaAdapter.py
class MediaAdapter:
  def __init__(self, moviePlayer):
    self.moviePlayer = moviePlayer

  def playMovie(self, filename):
    filename_detail = filename.lower().split('.')
    count_arr = len(filename_detail)

    if count_arr < 2:
      print("File not valid")
      return

    if filename_detail[count_arr - 1] == 'mkv':
      self.moviePlayer.playMKV(filename)
    elif filename_detail[count_arr - 1] == 'mp4':
      self.moviePlayer.playMP4(filename)
    else:
      print("File not supported")